Sources/Controllers/2.0/OACustomPOIViewController.h

```diff
@@ -14,7 +14,9 @@
 
 @required
 
-- (void) searchByUIFilter:(OAPOIUIFilter *)filter;
+- (void)searchByUIFilter:(OAPOIUIFilter *)filter wasSaved:(BOOL)wasSaved;
+- (BOOL)saveFilter:(OAPOIUIFilter *)filter alertDelegate:(id<UIAlertViewDelegate>)alertDelegate;
```

Comment:
Multiple issues here: why do you pass AlertViewDelegate here? UIAlertView is deprecated since ios 8.0, use UIAlertController instead. Lastly, ```updateRootScreen``` is a very confusing method, you should be able to achieve needed behavior without implementing such methods.